cmd/chain-maind/app/app.go

```diff
@@ -137,7 +137,24 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 		if err != nil {
 			return err
 		}
+
+		tmpFilePath := cleanedPath + ".tmp"
```

Comment:
how about genesisFilePath?
which tells what this file is

Reply:
cleanedPath + ".tmp"
this is ok, how about using sprintf formatting for readibility?

Reply:
Use WriteFile (github.com/google/renameio) to instead, no need to create a temp file